travis/travis_run_tests

```diff
@@ -10,13 +10,30 @@ from travis_helpers import success_msg, fail_msg
 
 
 def main(test_list):
+    """
+    Loop through each test and run them, add display results at the end
+
+    If the test has a .py extension, import as a list and call main function
+
+    :param list test_list: list of lists containing commands to run
+    :return: highest error code
+    """
     args = sys.argv[1:]
     results = []
     for test in test_list:
         # keep backward compatibility with version as an argument
         print("======== Testing %s ========" % test[0])
         test_w_args = test + args
-        res = subprocess.call(test_w_args)
+        test_file = test_w_args[0]
+        if test_file.endswith(".py"):
+            test_lib = test_file[:-3]
+            try:
+                res = __import__(test_lib).main(argv=test_w_args)
```

Comment:
This adds an implicit requirement that `.py` test files must have a `main()` function.
I don't see that as an issue but we should explicitly explain that somewhere (in comment code or in a README).